heudiconv/cli/run.py

```diff
@@ -137,6 +137,8 @@ def get_parser():
                        '(can be compressed) are supported in addition to '
                        'directory. All matching tarballs for a subject are '
                        'extracted and their content processed in a single pass')
+    group.add_argument('-r', '--recurse_dicom_dir', dest='recurse_dicom_dir',
+                       help='automatically recurse dicomdir sub directories to find all dicom files')
```

Comment:
note that it is relevant only in the case of using a template... I think the better and somewhat easier solution is just to support '**' globbing and demand python 3.5 for that.  `recursive` kwarg to glob could be set if '**' is detected in the glob pattern

Reply:
Hmm. I was under the impression that heudiconv was under python 2.7 support, thus this is the reason I chose to do it the way I did. Wouldn't switching to python 3.5 but more burdensome  overall than using perhaps a sub-optimal way to recurse through the dicom directories? 

Am I missing something?

Reply:
another option would be integrating [glob2](https://github.com/miracle2k/python-glob2) to support `**` globbing in py2/3

Reply:
I am in a bit of a time crunch for this. Could we merge the '--recurse_dicom_dir'?

Reply:
@tjhendrickson tests are not passing yet, but perhaps we don't need to add the additional argument - we can just make the recursive behavior "smart" in that if the `dicom_dir_template` contains `**`, we recurse. If not, we go about with a normal `glob` call

Reply:
Oh I see. I can look into this tomorrow.